app/src/main/java/com/alexbar/layoutbasic/screen/SnakeGameScreen.kt

```diff
@@ -0,0 +1,128 @@
+package com.alexbar.layoutbasic.screen
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.*
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.tooling.preview.Preview
+import com.alexbar.layoutbasic.ui.theme.*
+import com.alexbar.layoutbasic.utils.GameConstants
+import kotlinx.coroutines.delay
+import kotlin.random.Random
+
+@Composable
+fun SnakeGame() {
+    val snakeBodyColor = Color.Blue
+    val foodColor = Color.Red
+    val snakeSize = Dimens.snake_size_100
+    var snakePosition by remember { mutableStateOf(Offset.Zero) }
+    var direction by remember { mutableStateOf(Offset(snakeSize, Dimens.snake_init_direction_y)) }
+    var foodPosition by remember { mutableStateOf(randomOffset(snakeSize)) }
+    var gameOver by remember { mutableStateOf(false) }
+
+    LaunchedEffect(key1 = true) {
```

Comment:
I have changes alit bit this part like :
```
    LaunchedEffect(key1 = true) {
        while (!gameOver) {
            delay(GameConstants.game_delay_300_ms)
            val newPosition = snakePosition + direction
            when {
                newPosition.x < 0 || newPosition.y < 0 -> {
                    gameOver = true
                }
                snakePosition == foodPosition -> {
                    foodPosition = randomOffset(snakeSize)
                }
                else -> {
                    snakePosition = newPosition
                }
            }
        }
    }
```
    
Instead of using 'if-else-if' in your code, it's often better to use a 'when' structure. This makes your code easier to read and maintain